src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNodeReader.cs

```diff
@@ -554,17 +554,16 @@ public override void Close()
 
         public override string GetAttribute(int index)
         {
-            // https://github.com/dotnet/runtime/issues/44287
-            //     We should replace returning null with ArgumentOutOfRangeException
-            //     In case of not interactive state likely we should throw InvalidOperationException
             if (!IsInteractive)
             {
-                return null!;
+                throw new InvalidOperationException(SR.InvalidOperation_ExpectedInteractive);
```

Comment:
I do not really like this change, it might be the expected behavior but i see the other overloads above doing same check and still returning null.
https://github.com/dotnet/runtime/blob/32f5b72b2bde2aa6fcf4d0d202a898e9cb8233ce/src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNodeReader.cs#L468-L471
https://github.com/dotnet/runtime/blob/32f5b72b2bde2aa6fcf4d0d202a898e9cb8233ce/src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNodeReader.cs#L514-L517
1. i think it is breaking change
2. i think this behavior should be the same as the other overloads

Reply:
considering the AttributeCount returns 0: https://github.com/dotnet/runtime/blob/32f5b72b2bde2aa6fcf4d0d202a898e9cb8233ce/src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNodeReader.cs#L44

IMO we should also throw ArgumentOutOfRange. Other overloads do not throw because they don't throw I think

Reply:
(note that two examples you pasted return nullable so it makes sense to return null)

Reply:
> (note that two examples you pasted return nullable so it makes sense to return null)

Ah, didn't notice that, another option could be to return `string.Empty` which i see for other methods returning non null string, but i will let you decide what to do, overall i am OK with this PR changes
https://github.com/dotnet/runtime/blob/32f5b72b2bde2aa6fcf4d0d202a898e9cb8233ce/src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNodeReader.cs#L247-L252

Reply:
the namespace URI is a bit different since null and empty string are equivalent.

For the InvalidOperation vs ArgumentOutOfRange I'm ok either way. In the interactive mode you shouldn't be calling the method at all so InvalidOperation is accurate.